android/src/main/java/com/thunderhead/one_sdk_flutter/OnePlugin.java

```diff
@@ -29,132 +40,271 @@
-              .log(LogLevel.ERROR)
-              .log(Component.ANY)
-              .build();
+    private void sendResponseCode(String responseCode, String interactionPath, final Result result) {
+        executor.submit(() -> {
+            final OneResponseCodeRequest responseCodeRequest = new OneResponseCodeRequest.Builder()
+                    .responseCode(new OneResponseCode(responseCode))
+                    .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
+                    .build();
+
+            try {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                    One.sendResponseCode(responseCodeRequest);
+                    result.success(null);
+                } else {
+                    One.sendResponseCodeLegacySupport(responseCodeRequest);
+                    result.success(null);
+                }
+            } catch (ExecutionException error) {
+                result.error(Integer.toString(error.hashCode()), error.getLocalizedMessage(), error);
+                error.printStackTrace();
+            }
+        });
     }
 
-    One.setLoggingConfiguration(oneLoggingConfiguration);
-    result.success("Set Thunderhead LogLevel: " + (enabled ? "Enabled" : "Disabled"));
-  }
-
-  private void sendInteraction(String interactionPath, HashMap properties, final Result result) {
-    final OneRequest sendInteractionRequest = new OneRequest.Builder()
-            .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
-            .properties(properties)
-            .build();
-
-    final OneCall sendInteractionCall = One.sendInteraction(sendInteractionRequest);
-    sendInteractionCall.enqueue(new OneCallback() {
-      @Override
-      public void onSuccess(final OneResponse response) {
-        One.processResponse(response);
-        result.success(response.getTid());
-      }
-
-      @Override
-      public void onError(OneSDKError error) {
-        result.error(Integer.toString(error.getSystemCode()), error.getLocalizedMessage(), error);
-        Log.e(LOG_TAG, "SDK Error: " +  error.getErrorMessage());
-      }
-
-      @Override
-      public void onFailure(OneAPIError error) {
-        result.error(Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage(), error);
-        Log.e(LOG_TAG, "Api Error: " +  error.getErrorMessage());
-      }
-    });
-  }
+    private void optOut(MethodCall call, final Result result) {
+        OneOptOutConfiguration optOutConfiguration;
+
+        final HashMap<String, Object> arguments = call.arguments();
+        Boolean optOut = (Boolean) arguments.get("optOut");
+        List<String> options = (List<String>) arguments.get("options");
+
+        if (optOut == null) {
+            optOut = false;
+        }
+
+        if (options != null && !options.isEmpty()) {
+            for (String optOutOption : options) {
+                if (optOutOption.equals("cityCountryDetection")) {
```

Comment:
So if I am reading this correctly, this loop doesn't honor all options because you set a new configuration on each loop. You'll end up with whatever is last in the list.

Ex. if I do `One.optOut(true, [OneOptOptions.cityCountryDetection, OneOptOptions.otherOption])` in the future, this code will not honor both.